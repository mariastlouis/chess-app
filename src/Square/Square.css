.oneSquare {
  height: 100%;
  width: 100%;
  border: 1px solid black; }

.gray {
  background-color: gray; }

.white {
  background-color: white; }

.piece-container {
  height: 100px;
  width: 100px; }
