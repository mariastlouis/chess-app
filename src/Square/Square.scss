.oneSquare {
  height: 100%;
  width: 100%;
  border: 1px solid black;
}

.black {
  background-color: black;
}

.white {
  background-color: white;
}

.blue {
  background-color: blue;
}