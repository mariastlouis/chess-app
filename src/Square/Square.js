import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Square.css';
import Piece from '../Piece/Piece';
import Bishop from '../Bishop/Bishop';

class Square extends Component  {
  constructor () {
    super();
    this.state = {
      validMove: []
    }
  }

// = ({squareColor, knightPosition, bishopPosition, squareX, squareY, setSquare, selectPiece, chessPiece}) =>


checkColor() {
  const mode = this.props.gameMode;
  let modeColor;
  if (mode === 'standard') {
    modeColor = 'gray'
  } else if (mode === 'democrat'){
    modeColor = 'blue'
  } else {
      modeColor = 'red'
    }
  return this.reflectColor(modeColor)
}

  reflectColor(modeColor) {
    const color = this.props.squareColor ? 'white' : modeColor
    return color
  }

// checkColor() {
//   const mode = this.props.gameMode



//   const selectedColor = (mode) => {
//     let modeColor
//     if (mode === 'standard') {
//       modeColor = 'gray'
//     } else {
//       modeColor = 'blue'
//     }
//     console.log(modeColor)
//     return modeColor;
//   }


  

//   const color = this.props.squareColor ? 'white' : 'gray'
//   return color;
// }


displayPiece(){
   const [knightX, knightY] = this.props.knightPosition;
    const piece = (this.props.squareX === knightX && this.props.squareY === knightY) ?
    <Piece /> : null;
  return piece;
}

displayBishop (){
  const [bishopX, bishopY] = this.props.bishopPosition;
  const bishop = (this.props.squareX === bishopX && this.props.squareY === bishopY) ? 
  <Bishop /> : null;
  return bishop;
}

hoverSquare = event => {
  const id = event.target.id
  const piece = this.props.chessPiece
  if (piece === 'knight') {
    this.checkKnightMove(id)
  } 
}

// checkKnightMove(id) {
//   const arrayId = id.split("").map(Number)
//   const [x, y] = arrayId;
//   console.log(arrayId)
//   const [knightX, knightY] = this.props.knightPosition;
//   const moveX = Math.abs(x - knightX);
//   const moveY = Math.abs(y - knightY);
//    if((moveX === 2 && moveY === 1) || (moveX === 1 && moveY === 2) ){
//     this.setState({validMove: [x, y]})
//     console.log(this.state.validMove)
//      console.log('valid move')
//      this.validColor(id)

//     } else {
//      console.log('invalid move')
//     }

// }

// validColor(id){

// }

 //onMouseOver = {this.hoverSquare}>
  render () {
    const {squareColor, knightPosition, bishopPosition, squareX, squareY, setSquare, selectPiece, chessPiece} = this.props
  return (

    <div className = "Square">
      <div className = {`oneSquare ${this.checkColor()}`}>
        <div 
        className = "piece-container"
        id= {`${squareX}${squareY}`}
        onClick={ () => setSquare(squareX, squareY)}>     
          {this.displayPiece()}
          {this.displayBishop()}
        </div>   

      </div>
    </div>
  );
}
};

export default Square;