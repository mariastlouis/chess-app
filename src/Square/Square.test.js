import React from 'react';
import Square from './Square.js';
import {shallow} from 'enzyme';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

const mockSetSquareFunction = jest.fn();
const mockSelectPieceFunction = jest.fn();

let renderedSquare;

describe('Square', () => {
  beforeEach(() => {
    renderedSquare = shallow(
      <Square 
        squareColor = {true}
        knightPosition = {[1,7]}
        bishopPosition = {[2,7]}
        squareX = {3}
        squareY = {8}
        setSquare = {mockSetSquareFunction}
        selectPiece = {mockSelectPieceFunction}
        chessPiece = {'bishop'}
        gameMode = {'democrat'} />);
  })

  it('should render corrrectly', () => {
    expect(renderedSquare).toBeDefined();
  });

  it('should match the snapshot', () =>{
    expect(renderedSquare).toMatchSnapshot();
  });

it('should call the setSquare function if the square is clicked', () => {
    expect(mockSetSquareFunction.mock.calls.length).toEqual(0);
    renderedSquare.find('.piece-container').simulate('click');
    expect(mockSetSquareFunction.mock.calls.length).toEqual(1);
  });

it('should have a default class of white', () => {
  expect(renderedSquare.find('.white').length).toEqual(1);
});

});