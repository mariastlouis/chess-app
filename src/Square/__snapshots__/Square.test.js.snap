// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Square
    bishopPosition={
        Array [
            2,
            7,
          ]
    }
    chessPiece="bishop"
    gameMode="democrat"
    knightPosition={
        Array [
            1,
            7,
          ]
    }
    selectPiece={[Function]}
    setSquare={[Function]}
    squareColor={true}
    squareX={3}
    squareY={8}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="oneSquare white"
>
        <div
                className="piece-container"
                id="38"
                onClick={[Function]}
        >
                
        </div>
</div>,
      "className": "Square",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="piece-container"
          id="38"
          onClick={[Function]}
>
          
</div>,
        "className": "oneSquare white",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            null,
          ],
          "className": "piece-container",
          "id": "38",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          null,
          null,
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="oneSquare white"
>
          <div
                    className="piece-container"
                    id="38"
                    onClick={[Function]}
          >
                    
          </div>
</div>,
        "className": "Square",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="piece-container"
            id="38"
            onClick={[Function]}
>
            
</div>,
          "className": "oneSquare white",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              null,
            ],
            "className": "piece-container",
            "id": "38",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            null,
            null,
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
